pkg/util/validation/limits.go

```diff
@@ -88,6 +96,13 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.MaxGlobalSeriesPerMetric, "ingester.max-global-series-per-metric", 0, "The maximum number of active series per metric name, across the cluster. 0 to disable.")
 	f.IntVar(&l.MinChunkLength, "ingester.min-chunk-length", 0, "Minimum number of samples in an idle chunk to flush it to the store. Use with care, if chunks are less than this size they will be discarded.")
 
+	f.IntVar(&l.MaxMetricsWithMetadataPerQuery, "ingester.max-metrics-with-metadata-per-query", 4000, "The maximum number of metrics with metadata that a query can return.")
+	f.IntVar(&l.MaxMetadataPerQuery, "ingester.max-metadata-per-query", 5, "The maximum number of metadata per metric that a query can return.")
+	f.IntVar(&l.MaxLocalMetricsWithMetadataPerUser, "ingester.max-metadata-per-user", 8000, "The maximum number of active metrics with metadata per user, per ingester. 0 to disable.")
+	f.IntVar(&l.MaxLocalMetadataPerMetric, "ingester.max-metadata-per-metric", 10, "The maximum number of metadata per metric, per ingester. 0 to disable.")
+	f.IntVar(&l.MaxGlobalMetricsWithMetadataPerUser, "ingester.max-global-metadata-per-user", 0, "The maximum number of active metrics with metadata per user, across the cluster. 0 to disable. Supported only if -distributor.shard-by-all-labels is true.")
+	f.IntVar(&l.MaxGlobalMetadataPerMetric, "ingester.max-global-metadata-per-metric", 0, "The maximum number of metadata per metric, across the cluster. 0 to disable.")
```

Comment:
Local limits are scary like they're for series. There's no real good reason to use local limits, except that we're not able to enforce global limits if `shard-by-all-labels` is disabled. I'm wondering if could makes sense to always use `shard-by-all-labels` for metadata and remove local limits at all (supporting only global limits). @gouthamve @pstibrany ?